app/decorators/profile_decorator.rb

```diff
@@ -3,6 +3,12 @@ class ProfileDecorator < ApplicationDecorator
   # display area, e.g. :left_sidebar
   def ui_attributes_for(area:)
     names = ProfileField.public_send(area).pluck(:attribute_name)
-    data.slice(*names).select { |_, v| v.present? }
+    data.slice(*names)
+      .transform_keys { |k| label_for_attribute(k) }
+      .select { |k, v| k.present? && v.present? }
+  end
+
+  def label_for_attribute(attr)
+    ProfileField.find_by(attribute_name: attr)&.label
```

Comment:
Reading this again, I don't think we should do one DB request per attribute. Instead, we can probably remove the `pluck(:attribute_name)` from the first line of this method and then work from that, what do you think? Something like this:

```ruby
fields = ProfileField.public_send(area)
fields.to_h { |field| [field.label, data[field.attribute_name]] }.compact_blank
```